### PR TITLE
Implement Batch event Push

### PR DESCRIPTION
**Pull request by strider1981**
Implemented `ApplicationLifecycleListener` to register a `BroadcastReceiver` which listens to the Geofencing events and pushes data to Batch

### CODE REVIEW

File: android/src/main/java/expo/modules/geofencingbatchplugin/GeofencingEventsReceiver.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        private const val TAG = "GeofencingReceiver"
+    }
+
+    private val executorService: ExecutorService = Executors.newSingleThreadExecutor()
+
+    override fun onReceive(context: Context, intent: Intent) {
+        Log.d(TAG, "Received broadcast")
+        executorService.execute {
+            try {
+                Log.d(TAG, "executorService.execute")
```
Comment (sameerdhulap) [inline android/src/main/java/expo/modules/geofencingbatchplugin/GeofencingEventsReceiver.kt:28]: Data is different from what we specify in the woosmap documentation
https://developers.woosmap.com/products/geofencing-sdk/integration/batch/

File: android/src/main/java/expo/modules/geofencingbatchplugin/WoosmapExpoLifecycleListener.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+class WoosmapExpoLifecycleListener : ApplicationLifecycleListener {
+
+    companion object {
+        private const val TAG = "WoosmapExpoLifecycle"
+        private const val BATCH_API_KEY = "BATCH_API_KEY"
+    }
+
+    override fun onCreate(application: Application) {
+        super.onCreate(application)
+        startBatch(application)
```
Comment (sameerdhulap) [inline android/src/main/java/expo/modules/geofencingbatchplugin/WoosmapExpoLifecycleListener.kt]: `startBatch(application)` should not be part of our plugin; this responsibility lies with the batch SDK.

File: src/types.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export type ConfigProps = {
   apiKey: string;
+  batchApiKey: string;
```
Comment (sameerdhulap) [inline src/types.ts]: This should not be part of our configuration

**Review by sameerdhulap — CHANGES_REQUESTED** [review]

**Comment by strider1981** [comment]
@sameerdhulap This PR is still Work in Progress

**Pull request closed without merging**